src/Uno.UI/UI/Xaml/Controls/Repeater/ViewManager.cs

```diff
@@ -0,0 +1,984 @@
+				{
+					// Create one.
+					m_ElementFactoryRecycleArgs = new ElementFactoryRecycleArgs();
+				}
+
+				var context = m_ElementFactoryRecycleArgs;
+				context.Element = element;
+				context.Parent = m_owner;
+
+				m_owner.ItemTemplateShim.RecycleElement(context);
+
+				context.Element = null;
+				context.Parent = null;
+			}
+			else
+			{
+				// No ItemTemplate to recycle to, remove the element from the children collection.
+				var children = m_owner.Children;
+				int childIndex = 0;
+				childIndex = children.IndexOf(element);
+				if (childIndex < 0)
+				{
+					throw new InvalidOperationException("ItemsRepeater's child not found in its Children collection.");
+				}
+
+				children.RemoveAt(childIndex);
+			}
+
+			var virtInfo = ItemsRepeater.GetVirtualizationInfo(element);
+			virtInfo.MoveOwnershipToElementFactory();
+			m_phaser.StopPhasing(element, virtInfo);
+			if (m_lastFocusedElement == element)
+			{
+				// Focused element is going away. Remove the tracked last focused element
+				// and pick a reasonable next focus if we can find one within the layout 
+				// realized elements.
+				int clearedIndex = virtInfo.Index;
+				MoveFocusFromClearedIndex(clearedIndex);
+			}
+
+			REPEATER_TRACE_PERF("ElementCleared");
+		}
+
+		void MoveFocusFromClearedIndex(int clearedIndex)
+		{
+			UIElement focusedChild = null;
+			var focusCandidate = FindFocusCandidate(clearedIndex, focusedChild);
+			if (focusCandidate != null)
+			{
+				FocusState focusState = FocusState.Programmatic;
+				if (m_lastFocusedElement is Control focusedAsControl)
+				{
+					focusState = focusedAsControl.FocusState;
+				}
+
+				// If the last focused element has focus, use its focus state, if not use programmatic.
+				focusState = focusState == FocusState.Unfocused ? FocusState.Programmatic : focusState;
+				focusCandidate.Focus(focusState);
+
+				m_lastFocusedElement = focusedChild;
+				// Add pin to hold the focused element.
+				UpdatePin(focusedChild, true /* addPin */);
+			}
+			else
+			{
+				// We could not find a candiate.
+				m_lastFocusedElement = null;
+			}
+		}
+
+		Control FindFocusCandidate(int clearedIndex, UIElement focusedChild)
+		{
+			// Walk through all the children and find elements with index before and after the cleared index.
+			// Note that during a delete the next element would now have the same index.
+			int previousIndex = int.MinValue;
+			int nextIndex = int.MaxValue;
+			UIElement nextElement = null;
+			UIElement previousElement = null;
+			var children = m_owner.Children;
+			for (int i = 0; i < children.Count; ++i)
+			{
+				var child = children[i];
+				var virtInfo = ItemsRepeater.TryGetVirtualizationInfo(child);
+				if (virtInfo != null && virtInfo.IsHeldByLayout)
+				{
+					int currentIndex = virtInfo.Index;
+					if (currentIndex < clearedIndex)
+					{
+						if (currentIndex > previousIndex)
+						{
+							previousIndex = currentIndex;
+							previousElement = child;
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'previousElement'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)